app.js

```diff
@@ -0,0 +1,184 @@
+import express from 'express';
+import crypto from 'crypto';
+import { products } from './storage.js';
+import { carts } from './storage.js';
+import { users } from './storage.js';
+import { orders } from './storage.js';
+
+const app = express();
+const PORT = 3000;
+
+app.use(express.json());
+
+// register/create user
+app.post('/api/register', (req, res) => {
+    const { email, name, password } = req.body;
+
+    // unique user id is generated
+    const userId = crypto.randomUUID();
+
+    //regx email check user@example.com 
+    const correctEmail = /^[a-zA-Z0-9.-_]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+    
+    //check whether email format is correct. If it is wrong, return error;
+    if(!correctEmail.test(email)) {
+        return res.status(400).json({ error: 'Invalid email format' });
+    }
+
+    //check length of the email
+    if (email.length > 254) {
+        return res.status(400).json({ error: 'Email is too long (max 254 characters)' });
+    }
+
+    const correctPassword = /^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[!@#$%^&*])[A-Za-z\d!@#$%^&*]{8,}$/;
+
+    //check the password
+    if(!correctPassword.test(password)) {
+        return res.status(400).json({ error: 'Invalid email format' });
+    }
+
+    if (!email || !name || !password) {
+        return res.status(400).json({ error: 'Please put a correct email, name, password' });
+    }
+
+    const newUser = {id: userId, email, name, password };
+    users.push(newUser);
+    console.log(newUser);
+
+    res.status(200).json({
+        message: 'Your user is registered successfully',
+        user: {
+            id: userId,
+            email,
+            name,
+        }
+    });
+});
+
+//get all users
+app.get('/api/users', (req, res) => {
+    res.status(200).json(users);
+    console.log
+});
+
+// Middleware to check the header of the user and its validity
+function checkUserId(req, res, next) {
+    const userId = req.header('x-user-id');
+
+    if(!userId) {
+        return res.status(401).json({ error: 'Unauthorized: x-user-id header is required' });
+    }
+
+    const user = users.find(i => i.id === userId);
```

Comment:
`i` - bad name for the variable in this case

Reply:
Thank you, changed variables names here and in other similar cases